python.md

```diff
@@ -673,6 +674,21 @@ Using this technique, ensure the names accurately describe what the test is test
 This is less important for functional tests which don't call into a single
 object's API.
 
+### <a name="test-isolation">Isolation</a>
+
+Don't assume that tests that use the database are fully isolated from each other. Your
```

Comment:
I would start this sentence with "For transactional tests (ie one using `transaction=True`), don't assume...". Otherwise the sentence is misleading if you forget to read the next section.

Reply:
I think this is true for *all* tests though. If a transactional test runs concurrently with another, non-transactional test, when the transactional test commits, the other test will see the changes.

Reply:
I think the tests run with read-committed isolation (https://www.postgresql.org/docs/7.2/static/xact-read-committed.html) so they don't see writes from other tests (as they never commit unless in transaction mode).

Reply:
My understanding of read committed mode is that transactions see records from other transactions *as they are committed*.

"Also note that two successive SELECT commands can see different data, even though they are within a single transaction, if other transactions commit changes after the first SELECT starts and before the second SELECT starts."

Also: https://stackoverflow.com/questions/4034976/difference-between-read-commit-and-repeatable-read

I've also tried this by running two `shell_plus` sessions and it seems to support this.

Reply:
Let's talk about in person